src/tic_tac_toe_pair/game.clj

```diff
@@ -31,21 +41,23 @@
 
 (defn get-game-end-message [game]
   (let [winner (get-winning-token (:board game))]
-    (if winner
-      (build-congratulations-message winner)
-      "This game ended in a tie!")))
-
-(defn set-player-tokens [player-1-token game]
-  (assoc game 
-    :player-1-token player-1-token
-    :player-2-token (if (= :x player-1-token) :o :x)))
-  
-(defn play
-  ([game] (loop [history  [(set-player-tokens (get-player-mark game) game)]]
-  (let [game (last history)]
-    (if (is-game-over? game)
-      (do 
-        (draw-main game (get-game-end-message game))
-        history)
-        (let [new-game (update-game game (get-move-location game))]
-          (recur (conj history new-game))))))))
\ No newline at end of file
+    (cond 
+      (nil? winner) "This game ended in a tie!"
+      (= (:player-1-token game) winner) (build-congratulations-message winner)
+      :else (build-loosing-message winner))))
+      
+(defmulti get-move :current-token)
+(defmethod get-move :player-1-token [game]
```

Comment:
How would you refactor this so that either player could be computer or AI?

Reply:
Thanks for reviewing my PR Sarah!!!
This is the way I would refactor the function. What do you think? 

1. Change the game map so that players can be given a type value of :computer or :human like this:

{:player-1 {:token :o :type :computer}
  :player-2 {:token :x :type :human}
  :board [nil nil nil nil nil nil nil nil]
  :current-player :player-1
 }

2. Refactor the function so that it calls the correct function depending on the player type.

(defn move [game]
  (let [{{current-player-type :type} (:current-player game)} game] 
    (cond 
      (= :computer current-player-type) (get-ai-move)
      (= :human current-player-type) (get-move-location))))

Having players be either ai or human is not part of this story, so I will save this work on a separate branch for future use.

Reply:
That's like what I was thinking! I like giving the player both a marker and a type

Reply:
Great! I definitely plan on refactoring and implementing this.